sbt-plugin-test/build.sbt

```diff
@@ -182,10 +182,30 @@ lazy val jsDependenciesTest = project.settings(versionSettings: _*).
           "compressed/history.js appears before foo.js")
 
       streams.value.log.info("jsDependencies resolution test passed")
+
+      // Regression test for issue #2243
+      val pack = (packageJSDependencies in Compile).value
```

Comment:
Hem ... this code is in a task that `packageJSDependencies in Compile` depends on. You have a very weird circular dependency, here. I have no idea why sbt let you do that, but I don't think it's good to rely on it.

I suggest putting this code in a separate (custom) task.